### PR TITLE
CSS html generation

### PR DESCRIPTION
**Pull request by cloudRoutine**
This includes some cleanup of the project files and a new approach to generating code snippets.
Through CSS and html classes we can get popups without needing to use any js.
I've extended the amount of constructs that can be colored differently and eventually there will be a lot more options to easily customize the appearance of the snippets.

css-tooltips-styled

@matthid upgrading the FCS version managed to break the fsi evaluator, if you have any tips about how to fix it 👍

### CODE REVIEW

**Comment by matthid** [comment]
Actually when looking at this. Could we re-use something from ionide here? afaik it is also outputting html in the end, correct? (or is everything hidden behind code apis?)

**Comment by cloudRoutine** [comment]
Like @MangelMaxime was saying it'd be great to use a nice F# EDSL to knock out some templates, for the base functionality, [but now that I've looked into the available options a bit more](https://github.com/fsprojects/FSharp.Formatting/issues/469#issuecomment-390760333), I'm starting to think we should go our own way.

File: docs/files/content/style.css [hunk 1]
```diff
@@ -5,71 +5,71 @@
 /*--------------------------------------------------------------------------*/
 
 /* strings --- and stlyes for other string related formats */
-span.s { color:#E0E268; }
+span.s { color:#60815f; }
```
Comment (matthid) [inline docs/files/content/style.css]: We probably should leave that to someone else (I mean making F# code look nice). Ideally I'd like to steal/import code from ionide for example.

File: src/CSharpFormat/PaketFormat.cs [hunk 2]
```diff
@@ -4,78 +4,78 @@
 
 namespace Manoli.Utils.CSharpFormat
 {
-    /// <summary>
-    /// Generates color-coded Paket source code.
-    /// </summary>
-    public class PaketFormat : FSharpFormat
+	/// <summary>
```
Comment (matthid) [inline src/CSharpFormat/PaketFormat.cs]: tabs?

File: src/FSharp.CodeFormat/CssFormatting.fs [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+}
+"""   srcCssClass additionalCss
+
+
+
+/// Styles the popup that contains the intellisense content
+let tooltipPopup srcCssClass tipCssClass additionalCss =
+    sprintf  """
+/* Tooltip text */
+div .%s {
```
Comment (matthid) [inline src/FSharp.CodeFormat/CssFormatting.fs]: Can this stuff clash when you have lots of tooltips?

File: src/FSharp.Markdown/HtmlFormatting.fs [hunk 4]
```diff
@@ -49,24 +50,27 @@ type UniqueNameGenerator() =
             name
 
 /// Context passed around while formatting the HTML
-type FormattingContext =
-  { LineBreak : unit -> unit
+type HtmlFormattingContext = {
+    LineBreak : unit -> unit
```
Comment (matthid) [inline src/FSharp.Markdown/HtmlFormatting.fs]: Personally I prefer the old indentation, see https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting#formatting-record-declarations

File: tests/FSharp.Literate.Tests/EvalTests.fs [hunk 5]
```diff
@@ -50,21 +52,24 @@ printf ">>%d<<" 12343
...
+
+      // Contains transformed output
+      doc.Paragraphs |> shouldMatchPar (function
+        | CodeBlock ("42", _, _, _) -> true | _ -> false)
+      doc.Paragraphs |> shouldMatchPar (function
+        | CodeBlock ("85", _, _, _) -> true | _ -> false)
+      doc.Paragraphs |> shouldMatchPar (function
+        | CodeBlock (">>12343<<", _, _, _) -> true | _ -> false)
+    with e ->
+        Console.WriteLine doc
```
Comment (matthid) [inline tests/FSharp.Literate.Tests/EvalTests.fs]: ?
`reraise()`?

File: tests/FSharp.Markdown.Tests/CommonMarkSpecTest.fs [hunk 6]
```diff
@@ -48,14 +48,14 @@ let ``manual markdown test: show a blockquote with a code block`` () =
...
 
-	> ## This is a header.
-	> 
-	> 1.   This is the first list item.
-	> 2.   This is the second list item.
-	> 
-	> Here's some example code:
-	> 
-	>     return shell_exec("echo $input | $markdown_script");
+    > ## This is a header.
```
Comment (matthid) [inline tests/FSharp.Markdown.Tests/CommonMarkSpecTest.fs]: please revert this whitespace changes as it changes a tab-based-test

File: tests/FSharp.MetadataFormat.Tests/FSharp.MetadataFormat.Tests.fsproj [hunk 7]
```diff
@@ -7,32 +7,28 @@
...
   </ItemGroup>
-  
   <ItemGroup>
-    <ProjectReference Include="$(SolutionDir)\FSharp.Formatting.TestHelpers\FSharp.Formatting.TestHelpers.fsproj" />
-    <ProjectReference Include="$(SolutionDir)\src\FSharp.CodeFormat\FSharp.CodeFormat.fsproj" />
-    <ProjectReference Include="$(SolutionDir)\src\FSharp.Formatting.Common\FSharp.Formatting.Common.fsproj" />
-    <ProjectReference Include="$(SolutionDir)\src\FSharp.Formatting.Razor\FSharp.Formatting.Razor.fsproj" />
-    <ProjectReference Include="$(SolutionDir)\src\FSharp.Literate\FSharp.Literate.fsproj" />
-    <ProjectReference Include="$(SolutionDir)\src\FSharp.MetadataFormat\FSharp.MetadataFormat.fsproj" />
+    <ProjectReference Include="..\..\FSharp.Formatting.TestHelpers\FSharp.Formatting.TestHelpers.fsproj" />
```
Comment (matthid) [inline tests/FSharp.MetadataFormat.Tests/FSharp.MetadataFormat.Tests.fsproj]: why?

**Comment by wallymathieu** [comment]
Anything we can do to help out in order to merge this?

**Comment by cloudRoutine** [comment]
@wallymathieu I'm not too happy with this implementation anymore, and I've been writing a lot of CSS lately digging a bit deeper into what's possible just with CSS so I think I'll try a different approach once I have enough time to really dig into it in a few weeks.

On top of that there are so many conflicts with upstream changes merging would be a nightmare. 

But with F# 4.7 and a working netcore fsi, the time is definitely ripe to lock down a netcore implementation without any dependence on razor for templates and pure CSS output

**Pull request closed without merging**